### PR TITLE
issue-2594: use embedded for static files

### PR DESCRIPTION
**Pull request by dmitryk-dk**
https://github.com/VictoriaMetrics/VictoriaMetrics/issues/2594
https://github.com/VictoriaMetrics/VictoriaMetrics/issues/2652

### CODE REVIEW

File: app/vmagent/main.go [hunk 1]
```diff
@@ -322,11 +325,26 @@ func requestHandler(w http.ResponseWriter, r *http.Request) bool {
 			w.Write([]byte("OK"))
 		}
 		return true
+	default:
+		if strings.HasPrefix(r.URL.Path, "/static") {
+			file, err := staticFiles.ReadFile(strings.TrimPrefix(r.URL.Path, "/"))
```
Comment (hagen1778) [inline app/vmagent/main.go:330]: haven't checked that thoroughly, but looks like using `http.FileSystem` may simplify the logic. It also can autodetect headers based on content type, so you don't need to do it manually

**Review by f41gh7 — APPROVED** [review]
LGTM

File: app/vmagent/main.go [hunk 2]
```diff
@@ -322,11 +325,16 @@ func requestHandler(w http.ResponseWriter, r *http.Request) bool {
 			w.Write([]byte("OK"))
 		}
 		return true
+	default:
+		if strings.HasPrefix(r.URL.Path, "/static") {
+			http.FileServer(http.FS(staticFiles)).ServeHTTP(w, r)
```
Comment (f41gh7) [inline app/vmagent/main.go:330]: Probably, it's better to move into variable:

var staticServer = http.FileServer(http.FS(staticFiles))

 and use it later as:

staticServer.ServerHTTP(w,r)  

@hagen1778  WDYT?

Comment (hagen1778) [reply app/vmagent/main.go:330]: Yes, that looks reasonable.

**Comment by f41gh7** [comment]
Thanks!

**Pull request merged**